src/types.rs

```diff
@@ -833,6 +833,79 @@ impl<T: Float> From<MultiPolygon<T>> for Geometry<T> {
     }
 }
 
+impl<T: Float> Geometry<T> {
+
+    /// If this Geometry is a Point, then return that, else None.
+    ///
+    /// ```
+    /// use geo::*;
+    /// let g = Geometry::Point(Point::new(0., 0.));
+    /// let p2: Point<f32> = g.as_point().unwrap();
+    /// assert_eq!(p2, Point::new(0., 0.,));
+    /// ```
+    pub fn as_point(self) -> Option<Point<T>> {
+        if let Geometry::Point(x) = self {
+            Some(x)
+        } else {
+            None
+        }
+    }
+
+    /// If this Geometry is a LineString, then return that LineString, else None.
+    pub fn as_linestring(self) -> Option<LineString<T>> {
```

Comment:
i don't feel strongly about this, but should this be `as_line_string` since the struct is called a `LineString` (and not a `Linestring`). if we end up doing this, it would make `as_multi_line_string` a little verbose 😅